clients/src/main/java/org/apache/kafka/common/requests/DescribeShareGroupOffsetsResponse.java

```diff
@@ -17,21 +17,84 @@
 
 package org.apache.kafka.common.requests;
 
+import org.apache.kafka.common.TopicIdPartition;
 import org.apache.kafka.common.message.DescribeShareGroupOffsetsResponseData;
+import org.apache.kafka.common.message.DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseGroup;
+import org.apache.kafka.common.message.DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponsePartition;
+import org.apache.kafka.common.message.DescribeShareGroupOffsetsResponseData.DescribeShareGroupOffsetsResponseTopic;
 import org.apache.kafka.common.protocol.ApiKeys;
 import org.apache.kafka.common.protocol.ByteBufferAccessor;
 import org.apache.kafka.common.protocol.Errors;
 
 import java.nio.ByteBuffer;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class DescribeShareGroupOffsetsResponse extends AbstractResponse {
     private final DescribeShareGroupOffsetsResponseData data;
+    private final Map<String, Errors> groupLevelErrors = new HashMap<>();
 
     public DescribeShareGroupOffsetsResponse(DescribeShareGroupOffsetsResponseData data) {
         super(ApiKeys.DESCRIBE_SHARE_GROUP_OFFSETS);
         this.data = data;
+        for (DescribeShareGroupOffsetsResponseGroup group : data.groups()) {
+            this.groupLevelErrors.put(group.groupId(), Errors.forCode(group.errorCode()));
+        }
+    }
+
+    public DescribeShareGroupOffsetsResponse(int throttleTimeMs,
+                                             Map<String, Throwable> errorsMap,
+                                             Map<String, Map<TopicIdPartition, DescribeShareGroupOffsetsResponsePartition>> responseData) {
+        super(ApiKeys.DESCRIBE_SHARE_GROUP_OFFSETS);
+        List<DescribeShareGroupOffsetsResponseGroup> groupList = new ArrayList<>();
+        for (Map.Entry<String, Map<TopicIdPartition, DescribeShareGroupOffsetsResponsePartition>> groupEntry : responseData.entrySet()) {
+            String groupId = groupEntry.getKey();
+            Map<TopicIdPartition, DescribeShareGroupOffsetsResponsePartition> partitionDataMap = groupEntry.getValue();
```

Comment:
Can be replaced by a single line:
```suggestion
        responseData.forEach((groupId, partitionDataMap) -> {
```